### PR TITLE
Build/release fixes

### PR DESCRIPTION
**Pull request by bijanhoule**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Fix a few issues in the 0.2.0 release process.
- explicitly specify release settings on a few subprojects
- fix some cross-building issues
- move `controlApi` dependency to CLI project, it doesn't need to be published / marked as a spark dependency

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -227,7 +226,7 @@ lazy val apiDocs = (project in file("api"))
   .enablePlugins(OpenApiGeneratorPlugin)
   .settings(
     name := s"$artifactNamePrefix-docs",
-
+    skipReleaseSettings,
```
Comment (tdas) [inline build.sbt:229]: are all these explicit skipRelease settings needed?

File: project/ReleaseSettings.scala [hunk 2]
```diff
@@ -40,12 +40,16 @@ object ReleaseSettings {
     autoScalaLibrary := false,
   )
 
+  lazy val scalaReleaseSettings = releaseSettings ++ Seq(
+    crossPaths := true,
+    releaseCrossBuild := true,
+  )
+
   lazy val releaseSettings = Seq(
```
Comment (tdas) [inline project/ReleaseSettings.scala:48]: could this be made private. so that no one uses this directly?

File: build.sbt (hunk 1 above)
Comment (bijanhoule) [reply build.sbt:229]: I was running into failures without having them specifically added, yeah.  It looked like it was still attempting some part of the publish procuess but had only partial config

File: project/ReleaseSettings.scala (hunk 2 above)
Comment (bijanhoule) [reply project/ReleaseSettings.scala]: good call, added

**Review by tdas — APPROVED** [review]

**Pull request merged**